content/fast/reduce-network-payloads-using-text-compression/codelab-text-compression-brotli.md

```diff
@@ -0,0 +1,243 @@
+
+#### Cons:
++  Assets need to compressed with every build. Build times can increase significantly if high compression levels are used.
+
+#### Static compression with Node/Express and webpack
+
+Since **static compression** involves compressing files ahead of time, webpack settings can be modified to compress assets as part of the build step. The `brotli-webpack-plugin` can be used for this.
+
+Begin by adding it as a `devDependency` in `package.json`:
+
+<pre class="prettyprint">
+"devDependencies": {
+  //...
+ <strong>"brotli-webpack-plugin": "^1.1.0"</strong>
+},
+</pre>
+
+Like any other webpack plugin, import it in the configurations file, `webpack.config.js`:
+
+<pre class="prettyprint">
+const path = require("path");
+
+//...
+<strong>const BrotliPlugin = require('brotli-webpack-plugin');</strong>
+},
+</pre>
+
+And include it within the plugins array:
+
+<pre class="prettyprint">
+module.exports = {
+  // ... 
+  plugins: [
+    // ... 
+    <strong>new BrotliPlugin({
+      asset: '[file].br[query]',
+      test: /\.(js)$/
+    })</strong>
+  ]
+}
+},
+</pre>
+
+The following arguments are used in the plugin array:
++  `asset`: The target asset name. Defaults to `'[path].br[query]'`
+ +  `[file]` is replaced with the original asset file name
+ +  `[query]` is replaced with the query
++  `test`: All assets that match this RegExp (i.e. javascript assets ending in `.js`) are processed
+
+When the app reloads and rebuilds, a compressed version of the main bundle is now created. Open the Glitch Console to take a look at what's inside the final `public/` directory that's served by the Node server.
+
+<div class="aside note">
+The <code>public/</code> directory is included in the <code>.gitignore</code> file. Directories that contain build files are usually included here in order to be ignored by Git, and Glitch also hides these files from the editor tree.
+</div>
+
++ Click the **Status** button.
+
+<web-screenshot type="status"></web-screenshot>
+
++ Click the **Console** button.
+
+<web-screenshot type="console"></web-screenshot>
+
++ In the console, run the following commands to change into the `public` directory
+and see all of its files:
+
+<pre class="devsite-terminal devsite-click-to-copy">
+cd public
+ls -lh
+</pre>
+
+<img class="screenshot" src="./console-static-compression-brotli.png"  alt="Bundle size with static Brotli compression">
+
+The brotli compressed version of the bundle, `main.bundle.js.br`, is now saved here as well and is **~76% smaller in size** (225 KB vs. 53 KB) than `main.bundle.js`.
+
+Next, tell the server to send these brotli-compressed files whenever their original JS versions are being requested. This can be done by defining a new route in `server.js` before the files are served with `express.static`.
+
+<pre class="prettyprint">
+var express = require('express');
+
+var app = express();
+
+<strong>app.get('*.js', (req, res, next) => {
```

Comment:
Should this come after the calls to shrinkray and express.static?

Also, do we need to do anything to check that the client browser supports brotli?

Reply:
great catch, modified to check for br support by reading `accept-encoding` header.